### ISSUE TITLE
Not able to use Loopia provider with traefik

### ISSUE THREAD

**Issue by mingan666** [go-acme/lego#1542]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

Trying to setup the DNS challenge for Traefik and Loopia for a subdomain.

### What did you see instead?

Traefik is not able to generate the required TXT DNS entry.
It is not able to obtain a certificate for the host.
The traefik container log contains errors.

### How do you use lego?

Through Traefik

### Reproduction steps

Resolver related parts of the docker yml file:

```
  proxy:
    image: traefik:v2.5
    command:
      - "--certificatesresolvers.loopia.acme.dnschallenge=true"
      - "--certificatesresolvers.loopia.acme.dnschallenge.provider=loopia"
      - "--certificatesresolvers.loopia.acme.email=xxx"
      - "--certificatesresolvers.loopia.acme.storage=/letsencrypt/acme.json"
      - --log.level=DEBUG

    environment:
      - LOOPIA_API_USER_FILE=/run/secrets/loopia-api-user
      - LOOPIA_API_PASSWORD_FILE=/run/secrets/loopia-api-password

  whoami:
    image: traefik/whoami
    networks:
      - traefik_proxy
    deploy:
      labels:
        - traefik.enable=true
        - traefik.docker.network=traefik_proxy
        - traefik.http.routers.whoami.rule=Host(`xxx`)
        - traefik.http.routers.whoami.entrypoints=websecure
        - traefik.http.routers.whoami.tls.certresolver=loopia
        - traefik.http.services.whoami.loadbalancer.server.port=80

```       

### Version of lego

```console
Not sure where to find this?
Traefik version: "org.opencontainers.image.version": "v2.5.5"
According to: https://github.com/traefik/traefik/releases/tag/v2.5.5
[acme] Update go-acme/lego to v4.5.3
```

### Logs

<details>

```console
time="2021-12-12T16:19:20Z" level=debug msg="Loading ACME certificates [xxx]..." rule="Host(`xxx`)" providerName=loopia.acme routerName=whoami@docker
time="2021-12-12T16:19:20Z" level=debug msg="legolog: [INFO] [xxx] acme: Obtaining bundled SAN certificate"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/57862423640"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] acme: Could not find solver for: tls-alpn-01"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] acme: Could not find solver for: http-01"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] acme: use dns-01 solver"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] acme: Preparing to solve DNS-01"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [INFO] [xxx] acme: Cleaning DNS-01 challenge"
time="2021-12-12T16:19:21Z" level=debug msg="legolog: [WARN] [xxx] acme: cleaning up failed: loopia: failed to remove TXT record: unknown error: \"BAD_INDATA\" "
time="2021-12-12T16:19:21Z" level=error msg="Unable to obtain ACME certificate for domains \"xxx\": unable to generate a certificate for the domains [xxx]: error: one or more domains had a problem:\n[xxx] [xxx] acme: error presenting token: loopia: failed to add TXT record: unknown error: \"BAD_INDATA\"\n" providerName=loopia.acme routerName=whoami@docker rule="Host(`xxx`)"
```

</details>

### Go environment (if applicable)

<details>

```console
$ go version && go env
# paste output here
```

</details>

**Comment by ldez** [comment]
ping @hazzeh, could you take a look?

**Comment by ldez** [comment]
@mingan666 could you be more precise about your domain?
Instead of `xxx`, could use something more representative of your domain (ex: `xxx.example.com` or `example.com`)?
I don't ask for your domain but just something that can give us more information about your context.

**Comment by mingan666** [comment]
Its a subdomain to my swedish domain through Loopia.
So replacing my domain with example it would be:
`whoami.example.se`

**Comment by mingan666** [comment]
I've both tried with a subdomain already defined in loopia dns and one that does not exist.
Both fail.

**Comment by ldez** [comment]
Could you also check that the `LOOPIA_API_USER_FILE` and `LOOPIA_API_PASSWORD_FILE` don't contain extra characters (like new line or white space, ...)?

**Comment by ldez** [comment]
is your domain contain non-ASCII characters?

**Comment by mingan666** [comment]
I logged into the container and looked at the secret files. They look fine to me.
Its just normal characters in the domain.

**Comment by ldez** [comment]
"normal characters" in Swedish or ASCII?

for example `åäöø` are not ASCII characters.

**Comment by ldez** [comment]
I ask for that because your error:
> loopia: failed to add TXT record: unknown error: \"BAD_INDATA\"

> `BAD_INDATA` Some parameter has invalid values.

https://www.loopia.com/api/status/

The parameters are the username, the password, the TXT value, the domain, and the sub-domain.

**Comment by mingan666** [comment]
Swedish characters (äöå) are not normal characters hihi
My domain has only normal characters (a-z).
Since you asked I did try the link above and as I expected the result was the same.

I did change to a password without special characters (A-Z, a-z, 0-9)
The usernames contain a @-sign though, but I think all api user end in @loopiaapi?

**Comment by hazzeh** [comment]
> ping @hazzeh, could you take a look?
> 
> https://www.loopia.com/api/status/

Sure, I'll look into it.

**Comment by hazzeh** [comment]
I have not been able to reproduce it, but will continue testing.
I created a simple docker swarm deployment, since I assume that is what you have. The secrets were created in advance.

```yaml
version: "3.3"
services:
  traefik:
    image: "traefik:v2.5"
    command:
      - "--providers.docker=true"
      - "--certificatesresolvers.loopia.acme.dnschallenge=true"
      - "--certificatesresolvers.loopia.acme.dnschallenge.provider=loopia"
      - "--certificatesresolvers.loopia.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory"
      - "--certificatesresolvers.loopia.acme.dnschallenge.resolvers=1.1.1.1:53,8.8.8.8:53"
      - "--certificatesresolvers.loopia.acme.email=xxx@example.com"
    environment:
      - LOOPIA_API_USER_FILE=/run/secrets/loopia-api-user
      - LOOPIA_API_PASSWORD_FILE=/run/secrets/loopia-api-password
    secrets:
      - loopia-api-user
      - loopia-api-password
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

  whoami:
    image: "traefik/whoami"
    labels:
      - "traefik.http.routers.whoami.rule=Host(`whoami.example.com`)"
      - "traefik.http.routers.whoami.tls.certresolver=loopia"
secrets:
  loopia-api-user:
    external: true
  loopia-api-password:
    external: true
```

Can you try if that compose file works for you?

**Comment by mingan666** [comment]
Your compose file worked after updating to the correct domain values.
After doing some trail and error I also got my compose file to work by adding the following:

`      - "--certificatesresolvers.loopia.acme.dnschallenge.resolvers=1.1.1.1:53,8.8.8.8:53"`

I did not find any examples with this parameter in the [traefik user guide](https://doc.traefik.io/traefik/user-guides/docker-compose/acme-dns/) and do not really understand why it is needed for the traefik-loopia combination. Or could it be related to my local network settings?

Maybe an update of the lego loopia documentation is warranted?

Anyway, thanks a lot for you help!
